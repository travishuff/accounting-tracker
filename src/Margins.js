import React from 'react'

import { Row, Col, FormGroup, FormInput } from 'shards-react'

class Margins extends React.Component {
  render() {
    const {
      buyPrice,
      handleBuyPriceChange,
      sellPrice,
      handleSellPriceChange,
    } = this.props

    return (
      <React.Fragment>
        <Row className="row-title">
          <Col>Experiment with your margins!</Col>
        </Row>
        <Row>
          <Col>
            <FormGroup>
              <label htmlFor="buy">
                Buy Price
                <FormInput
                  type="number"
                  name="buy"
                  value={buyPrice}
                  onChange={handleBuyPriceChange}
                  onBlur={handleBuyPriceChange}
                  size="sm"
                />
              </label>
            </FormGroup>
            <FormGroup>
              <label htmlFor="sell">
                Sell Price
                <FormInput
                  type="number"
                  name="sell"
                  value={sellPrice}
                  onChange={handleSellPriceChange}
                  onBlur={handleSellPriceChange}
                  size="sm"
                />
              </label>
            </FormGroup>
          </Col>
        </Row>
      </React.Fragment>
    )
  }
}

export default Margins
