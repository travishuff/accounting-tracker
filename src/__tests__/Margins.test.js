import React from 'react'
import { create } from 'react-test-renderer'

import Margins from '../Margins'

// Snapshot test
test('snapshot', () => {
  const c = create(<Margins />)

  expect(c.toJSON()).toMatchSnapshot()
})
