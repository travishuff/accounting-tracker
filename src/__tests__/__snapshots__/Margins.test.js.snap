// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <h3>
    Experiment with your margins!
  </h3>,
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="fields"
    >
      <div
        className="four wide field"
      >
        <label>
          Buy Price:
        </label>
        <div
          className="ui input"
        >
          <input
            name="buyPrice"
            onChange={[Function]}
            type="number"
          />
        </div>
      </div>
      <div
        className="four wide field"
      >
        <label>
          Sell Price:
        </label>
        <div
          className="ui input"
        >
          <input
            name="sellPrice"
            onChange={[Function]}
            type="number"
          />
        </div>
      </div>
    </div>
  </form>,
]
`;
